### PR TITLE
kata-deploy: Add drop-in conf support for containerd

### PR DESCRIPTION
**Pull request by zvonkok**
Upcoming containerd 2.0 will have drop-in conf support. First patch to get this functionality into our stack.

### CODE REVIEW

File: tools/packaging/kata-deploy/scripts/kata-deploy.sh [hunk 1]
```diff
@@ -401,7 +402,8 @@ function configure_containerd() {
 
 	mkdir -p /etc/containerd/
 
-	if [ -f "$containerd_conf_file" ]; then
+	# If we have a drop-in configuration file, we don't need to touch the original
+	if [ -f "$containerd_conf_file" ]&& [ -z "${CONTAINERD_DROP_IN_CONF}" ]; then
```
Comment (c3d) [inline tools/packaging/kata-deploy/scripts/kata-deploy.sh:406]: Minor nit
```suggestion
	if [ -f "$containerd_conf_file" ] && [ -z "${CONTAINERD_DROP_IN_CONF}" ]; then
```

**Review by c3d — APPROVED** [review]
Except for a minor nit, looks good to me.

**Comment by sprt** [comment]
Do you have a reference for this? Would like to learn more about this containerd feature.

**Comment by zvonkok** [comment]
@sprt There is some work being done on the proper merge behaviour: https://github.com/containerd/containerd/pull/9982

**Review by BbolroC — APPROVED** [review]
LGTM, thanks @zvonkok !

**Comment by fidencio** [comment]
> Upcoming containerd 2.0 will have drop-in conf support. First patch to get this functionality into our stack.

One question here, you want to let the user decide what to do?
Can we check for whether user is running 1.7 or higher (as this was actually introduced in 1.7), and then just manually add the imports (in case those are not there), and drop-in the files regardless?

Another thing that's crossing my mind is how we differentiate v1 / v2 / v3 when there's no configuration set, and drop-ins are used.

**Pull request closed without merging**